experiments/benchmarks/land.jl

```diff
@@ -0,0 +1,670 @@
+            dss! = ClimaLand.dss!,
+        ),
+        Y,
+        (t0, tf),
+        p,
+    )
+
+    updateat = Array(t0:(3600 * 3):tf)
+    updatefunc = ClimaLand.make_update_drivers(atmos, radiation)
+    cb = ClimaLand.DriverUpdateCallback(updateat, updatefunc)
+    return prob, cb
+end
+
+function setup_and_solve_problem(; greet = false)
+    # We profile the setup phase as well here. This is not intended, but it is the easiest
+    # to set up for both CPU/GPU at the same time
+    t0 = 0.0
+    tf = 60 * 60.0 * 3
+    Δt = 180.0
+    nelements = (101, 15)
+    if greet
+        @info "Run: Global RichardsModel"
+        @info "Resolution: $nelements"
+        @info "Timestep: $Δt s"
+        @info "Duration: $(tf - t0) s"
+    end
+
+    prob, cb = setup_prob(t0, tf, Δt; nelements)
+
+    # Define timestepper and ODE algorithm
+    stepper = CTS.ARS343()
+    norm_condition = CTS.MaximumError(FT(1e-8))
+    conv_checker = CTS.ConvergenceChecker(; norm_condition = norm_condition)
+    ode_algo = CTS.IMEXAlgorithm(
+        stepper,
+        CTS.NewtonsMethod(
+            max_iters = 1,
+            update_j = CTS.UpdateEvery(CTS.NewNewtonIteration),
+            convergence_checker = conv_checker,
+        ),
+    )
+
+    SciMLBase.solve(prob, ode_algo; dt = Δt, callback = cb)
+    return nothing
+end
+
+# Warm up and greet
+setup_and_solve_problem(; greet = true)
+
+@info "Starting profiling"
+# Stop when we profile for MAX_PROFILING_TIME_SECONDS or MAX_PROFILING_SAMPLES
+MAX_PROFILING_TIME_SECONDS = 500
+MAX_PROFILING_SAMPLES = 100
+time_now = time()
+timings_s = Float64[]
+while (time() - time_now) < MAX_PROFILING_TIME_SECONDS &&
+    length(timings_s) < MAX_PROFILING_SAMPLES
+    push!(timings_s, ClimaComms.@elapsed device setup_and_solve_problem())
+end
+num_samples = length(timings_s)
+average_timing_s = round(sum(timings_s) / num_samples, sigdigits = 3)
+max_timing_s = round(maximum(timings_s), sigdigits = 3)
+min_timing_s = round(minimum(timings_s), sigdigits = 3)
+std_timing_s = round(
+    sum(((timings_s .- average_timing_s) .^ 2) / num_samples),
+    sigdigits = 3,
+)
+@info "Num samples: $num_samples"
+@info "Average time: $(average_timing_s) s"
+@info "Max time: $(max_timing_s) s"
+@info "Min time: $(min_timing_s) s"
+@info "Standard deviation time: $(std_timing_s) s"
+@info "Done profiling"
+
+Profile.@profile setup_and_solve_problem()
+results = Profile.fetch()
+flame_file = joinpath(outdir, "flame_$device_suffix.html")
+ProfileCanvas.html_file(flame_file, results)
+@info "Saved compute flame to $flame_file"
+
+Profile.Allocs.@profile sample_rate = 0.01 setup_and_solve_problem()
+results = Profile.Allocs.fetch()
+profile = ProfileCanvas.view_allocs(results)
+alloc_flame_file = joinpath(outdir, "alloc_flame_$device_suffix.html")
+ProfileCanvas.html_file(alloc_flame_file, profile)
+@info "Saved allocation flame to $alloc_flame_file"
+
+if ClimaComms.device() isa ClimaComms.CUDADevice
```

Comment:
@Sbozzolo do you know what this extra CUDA profile call does? It doesnt upload anything. When we run in buildkite, it successfully got through line 650, and then ran out of memory.

Reply:
It should provide detailed information about the cost of kernels, but we can remove it if it is a problem (we can put it back when things are more optimized)